lib/models/users-schema.js

```diff
@@ -17,38 +17,54 @@ const users = new mongoose.Schema({
     type: String,
     required: true,
     default: 'user',
-    enum: ['admin', 'editor', 'user']
-  }
+    enum: ['admin', 'editor', 'user'],
+  },
 });
 
-// TODO JSDocs Comment
+/**
+ * Pre save is a function which allows us to check if a user has modified their password. If they have we resave it in hashed format.
+ * 
+ */
 users.pre('save', async function() {
-  // TODO README Question
-  // What does .isModified do and why do we use it?
   if (this.isModified('password')) {
     this.password = await bcrypt.hash(this.password, 10);
   }
 });
 
 // TODO JSDocs Comment
-// TODO Convert this function into using async/await
-users.statics.authenticate = function(creds) {
+/**
+ * Users.authenticate is a functiomn that allows us to find a user based off of passed in credentails
+ * Those credentials are then varified. 
+ * @param {Object} creds - The user credentials we are trying to authenticate
+ * @returns {Boolean} - If the user and user password match return true else return false
```

Comment:
What you are actually returning will be the `{Promise<object>}` of a record/user that was successfully authenticated